src/MongoDB/Driver/Server.cpp

```diff
@@ -138,169 +162,237 @@ Array HHVM_METHOD(MongoDBDriverServer, __debugInfo)
 }
 
 int64_t HHVM_METHOD(MongoDBDriverServer, getLatency)
 {
 	MongoDBDriverServerData* data = Native::data<MongoDBDriverServerData>(this_);
 	mongoc_server_description_t *sd;
-	bson_error_t error;
 
-	if ((sd = mongoc_topology_description_server_by_id(&data->m_client->topology->description, data->m_server_id, &error))) {
-		return (int64_t) (sd->round_trip_time);
+	if ((sd = mongoc_client_get_server_description(data->m_client, data->m_server_id))) {
+		int64_t round_trip;
+
+		round_trip = mongoc_server_description_round_trip_time(sd);
+
+		mongoc_server_description_destroy(sd);
+
+		return round_trip;
 	}
 
-	throw MongoDriver::Utils::CreateAndConstruct(MongoDriver::s_MongoDriverExceptionRuntimeException_className, "Failed to get server description, server likely gone: " + HPHP::Variant(error.message).toString(), HPHP::Variant((uint64_t) 0));
+	throw MongoDriver::Utils::CreateAndConstruct(MongoDriver::s_MongoDriverExceptionRuntimeException_className, "Failed to get server description", HPHP::Variant((uint64_t) 0));
 }
 
 int64_t HHVM_METHOD(MongoDBDriverServer, getPort)
 {
 	MongoDBDriverServerData* data = Native::data<MongoDBDriverServerData>(this_);
 	mongoc_server_description_t *sd;
-	bson_error_t error;
 
-	if ((sd = mongoc_topology_description_server_by_id(&data->m_client->topology->description, data->m_server_id, &error))) {
-		return (int64_t) (sd->host.port);
+	if ((sd = mongoc_client_get_server_description(data->m_client, data->m_server_id))) {
+		int64_t port;
+
+		port = mongoc_server_description_host(sd)->port;
+		mongoc_server_description_destroy(sd);
+
+		return port;
 	}
 
-	throw MongoDriver::Utils::CreateAndConstruct(MongoDriver::s_MongoDriverExceptionRuntimeException_className, "Failed to get server description, server likely gone: " + HPHP::Variant(error.message).toString(), HPHP::Variant((uint64_t) 0));
+	throw MongoDriver::Utils::CreateAndConstruct(MongoDriver::s_MongoDriverExceptionRuntimeException_className, "Failed to get server description", HPHP::Variant((uint64_t) 0));
 }
 
 Array HHVM_METHOD(MongoDBDriverServer, getTags)
 {
 	MongoDBDriverServerData* data = Native::data<MongoDBDriverServerData>(this_);
 	mongoc_server_description_t *sd;
-	bson_error_t error;
 
-	if ((sd = mongoc_topology_description_server_by_id(&data->m_client->topology->description, data->m_server_id, &error))) {
-		Variant v;
+	if ((sd = mongoc_client_get_server_description(data->m_client, data->m_server_id))) {
+		const bson_t       *is_master = mongoc_server_description_ismaster(sd);
+
+		Variant v_last_is_master;
+		Array   a_last_is_master;
+
 		hippo_bson_conversion_options_t options = HIPPO_TYPEMAP_DEBUG_INITIALIZER;
 
-		BsonToVariantConverter convertor(bson_get_data(&sd->tags), sd->tags.len, options);
-		convertor.convert(&v);
+		BsonToVariantConverter convertor(bson_get_data(is_master), is_master->len, options);
+		convertor.convert(&v_last_is_master);
```

Comment:
Same question as https://github.com/mongodb/mongo-hhvm-driver/pull/94/files#r61436012 for `getInfo()`. If conversion throws, will this leak?